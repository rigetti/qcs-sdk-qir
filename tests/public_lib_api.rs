use qcs_sdk_qir::transpile_qir_to_quil;
use std::fs::read;

#[test]
fn transpile_qir_to_quil_bell_state() {
    let data = read("tests/fixtures/programs/bell_state.bc").unwrap();
    let output = transpile_qir_to_quil(&data).unwrap();
    insta::assert_snapshot!(output.program.to_string(true));
    insta::assert_display_snapshot!(output.shot_count);
}
