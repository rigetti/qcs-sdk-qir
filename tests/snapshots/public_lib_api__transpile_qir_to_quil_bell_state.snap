---
source: tests/public_lib_api.rs
expression: output.program.to_string(true)
---
DECLARE ro BIT[2]
H 0
CNOT 0 1
MEASURE 0 ro[0]
MEASURE 1 ro[1]

